CRM/Core/Form.php

```diff
@@ -2888,6 +2889,15 @@ public function getSubmittedValue(string $fieldName) {
     if (in_array($fieldName, $this->submittableMoneyFields, TRUE)) {
       return CRM_Utils_Rule::cleanMoney($value);
     }
+    else {
+      $fieldRules = $this->_rules[$fieldName] ?? [];
+      foreach ($fieldRules as $rule) {
```

Comment:
This is really clever - I think we could replace the `submittableMoneyFields` with a similar concept around the money rule